### ISSUE TITLE
No RTX no CogStudio?

### ISSUE THREAD

**Issue by merket** [pinokiofactory/cogstudio#14]
I was so excited to see that this app would run with low VRAMs under 10GB.
But I believe unless the GPU is RTX architecture we can not use it right?

Can somebody confirm or deny this?

Because when I try to run it I get an instant error of "can not allocate 35GB" for my 8gb card...
Somebody on discord said that this is an issue related to not having RTX.

**Comment by ondrovic** [comment]
I don't know I have an 3090 RTX and it working fine

![image](https://github.com/user-attachments/assets/dc39ed78-c19b-4ff5-b30a-079c635c97f3)

**Comment by shutarojp** [comment]
> I don't know I have an 3090 RTX and it working fine
> 
> ![image](https://private-user-images.githubusercontent.com/3682042/369976828-dc39ed78-c19b-4ff5-b30a-079c635c97f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDc1OTksIm5iZiI6MTcyNzMwNzI5OSwicGF0aCI6Ii8zNjgyMDQyLzM2OTk3NjgyOC1kYzM5ZWQ3OC1jMTliLTRmZjUtYjMwYS0wNzljNjM1Yzk3ZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMjMzNDU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjVmZGM4ZTA3NzMzNmJiYzg5MzNmOTdjOTc1MzQ2MzE0ZDE3ZmU4MjkyY2JlNTU1ZjdjMGU0MjE2NDc1NjM1NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.OSheN3F0ve--SVZupqFSDr42mFv4JnDfxEQEL7tZ4V4)

but 3090 got 24G Vram, this only take under 5G Vrams.

**Comment by tw9mini** [comment]
I'm using comfyUI so not sure if it applies here, but do you have cpu offloading, tiling and slicing enabled?

**Comment by shutarojp** [comment]
> > I don't know I have an 3090 RTX and it working fine
> > ![image](https://private-user-images.githubusercontent.com/3682042/369976828-dc39ed78-c19b-4ff5-b30a-079c635c97f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNjUxMDYsIm5iZiI6MTcyNzM2NDgwNiwicGF0aCI6Ii8zNjgyMDQyLzM2OTk3NjgyOC1kYzM5ZWQ3OC1jMTliLTRmZjUtYjMwYS0wNzljNjM1Yzk3ZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMTUzMzI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTFhMGRlMDMxMjVjODE3MzhlM2YxZjkyNTRmYzUyOTM5YmViOTI2ODZjMGU5MTZlNjZlN2FhNWQxZWNiMTY5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.DChuW6dl4exaHocx9w78mCGCmAp8_KjAfAY5PUCXERQ)
> 
> I believe you wanted to show that it only uses 5GB so it should've worked on mine as well, but as I mentioned in my OP; someone on discord said that on GPUs without RTX it falsely tries to allocate wrong amount of VRAM to the gpu... So If I had a gpu with RTX and only 8gbs of ram, may be it would've tried to allocate only 3-5 gbs... but without RTX architecture the script goes bonkers.

Yes, even I have 3090 my card with 24 VRAM, but still less 5g vrams only.

**Comment by shutarojp** [comment]
> > > > I don't know I have an 3090 RTX and it working fine
> > > > ![image](https://private-user-images.githubusercontent.com/3682042/369976828-dc39ed78-c19b-4ff5-b30a-079c635c97f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNjUxMDYsIm5iZiI6MTcyNzM2NDgwNiwicGF0aCI6Ii8zNjgyMDQyLzM2OTk3NjgyOC1kYzM5ZWQ3OC1jMTliLTRmZjUtYjMwYS0wNzljNjM1Yzk3ZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMTUzMzI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTFhMGRlMDMxMjVjODE3MzhlM2YxZjkyNTRmYzUyOTM5YmViOTI2ODZjMGU5MTZlNjZlN2FhNWQxZWNiMTY5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.DChuW6dl4exaHocx9w78mCGCmAp8_KjAfAY5PUCXERQ)
> > > 
> > > 
> > > I believe you wanted to show that it only uses 5GB so it should've worked on mine as well, but as I mentioned in my OP; someone on discord said that on GPUs without RTX it falsely tries to allocate wrong amount of VRAM to the gpu... So If I had a gpu with RTX and only 8gbs of ram, may be it would've tried to allocate only 3-5 gbs... but without RTX architecture the script goes bonkers.
> > 
> > 
> > Yes, even I have 3090 my card with 24 VRAM, but still less 5g vrams only.
> 
> Well, thank you for your input, but as I described there seems to be an allocation problem with the cards which do not have RTX and that is the whole problem. Meaning that even if the job needed less than 8gbs it tries to allocate much more than needed to VRAM in old cards.
> 
> At least that is what they told me on discord, I was just trying to confirm that, because I would very much to be able to use CogStudio on my non RTX card with low vram, as well.

I suggest that they add the option of using the percentage of VRAM of the graphics card or how much VRAM to use, so that everyone can set it according to their own graphics card.

**Comment by Seedmanc** [comment]
I have 2070S RTX and the part about only 5 gigs is a lie, it crashes with CUDA OOM.
